lib/app_web/controllers/api_controller.ex

```diff
@@ -3,23 +3,37 @@ defmodule AppWeb.ApiController do
   require Logger
 
   def create(conn, %{"" => params}) do
-    # dbg(params)
-    # check if content_type e.g: "image/png"
+
+    # Check if content_type e.g: "image/png"
     if String.contains?(params.content_type, "image") do
-      try do
-        {:ok, body} = App.Upload.upload(params)
-        render(conn, :success, body)
-      rescue
-        e ->
-          Logger.error(Exception.format(:error, e, __STACKTRACE__))
-          render(conn |> put_status(400), %{body: "Error uploading file #26"})
+      case App.Upload.upload(params) do
+        {:ok, body} ->
+          render(conn, :success, body)
+
+        {:error, :failure_read} ->
+          render(conn |> put_status(400), %{body: "Error uploading file. Failure reading file."})
+
+        {:error, :invalid_cid} ->
+          render(conn |> put_status(400), %{
+            body: "Error uploading file. Failure creating the CID filename."
```

Comment:
This error message means _nothing_ to the person using the `API`. 
They don't know (_or **care**_) that we'er using `CIDs` under the hood.
This is worse than a generic error message because:
a) It doesn't log anything so we cannot debug it.
b) it returns information the `API` caller doesn't know what to _do_ with.

How does the caller of the `API` modify their request in response to this error? 🤷‍♂️

Reply:
I've made the changes for adding logging and returning information only on the relevant scenarios where the person can change (no file content, invalid content type, empty image files).

Given that you've made the changes for the relevant pattern matches to be returned to the person in [`ef113ec` (#86)](https://github.com/dwyl/imgup/pull/86/commits/ef113ec4d749fe0744909d5d8c74d538c5acccfe), I'm marking this as resolved as well.